geomstats/geometry/heisenberg.py

```diff
@@ -0,0 +1,225 @@
+        """
+        return gs.zeros(self.dim)
+
+    identity = property(get_identity)
+
+    def belongs(self, point):
+        """Evaluate if a point belongs to the 3D Heisenberg group.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 3]
+            Point to evaluate.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the Heisenberg
+            group (i.e. R^3 in this representation).
+        """
+        point_dim = point.shape[-1]
+        belongs = point_dim == self.dim
+        if gs.ndim(point) == 2:
+            belongs = gs.tile([belongs], (point.shape[0],))
+
+        return belongs
+
+    def compose(self, point_a, point_b):
+        """Compute the group product of elements `point_a` and `point_b`.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., 3]
+            Left factor in the product.
+        point_b : array-like, shape=[..., 3]
+            Right factor in the product.
+
+        Returns
+        -------
+        composed : array-like, shape=[..., 3]
+            Product of point_a and point_b along the first dimension.
+        """
+        point_ab = point_a + point_b
+        point_ab_additional_term = gs.array(1 / 2 * (point_a[..., 0] *
+                                                     point_b[..., 1] - point_a[..., 1] *
+                                                     point_b[..., 0]))
+
+        point_ab = point_ab + gs.concatenate([gs.zeros_like(point_ab[..., :2]),
+                                              point_ab_additional_term[..., None]],
+                                             axis=-1)
+
+        return point_ab
+
+    def inverse(self, point):
+        """Compute the group inverse of point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 3]
+            Point.
+
+        Returns
+        -------
+        inv_point : array-like, shape=[..., 3]
+            Inverse.
+        """
+        return -point
+
+    def jacobian_translation(
+            self, point, left_or_right='left'):
+        """Compute the Jacobian matrix of left/right translation by a point.
+
+        This calculates the differential of the left translation L_(point)
+        evaluated at 'point'. Note that it only depends on the point we are
+        left-translating by, not on the point where the differential is evaluated.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 3]
+            Point.
+        left_or_right : str, {'left', 'right'}
+            Indicate whether to calculate the differential of left or right
+            translations.
+            Optional, default: 'left'.
+
+        Returns
+        -------
+        jacobian : array-like, shape=[..., 3, 3]
+            Jacobian of the left/right translation by point.
+        """
+        E31 = gs.array_from_sparse([(2, 0)], [1.], (3, 3))
```

Comment:
No uppercase letters in variable names, see:
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names